libhoney.go

```diff
@@ -133,40 +133,44 @@ type Config struct {
 // VerifyWriteKey calls out to the Honeycomb API to validate the write key, so
 // we can exit immediately if desired instead of happily sending events that
 // are all rejected.
-func VerifyWriteKey(config Config) error {
+func VerifyWriteKey(config Config) (string, error) {
 	if config.WriteKey == "" {
-		return errors.New("Write key is empty")
+		return "", errors.New("Write key is empty")
 	}
 	if config.APIHost == "" {
 		config.APIHost = defaultAPIHost
 	}
 	u, err := url.Parse(config.APIHost)
 	if err != nil {
-		return fmt.Errorf("Error parsing API URL: %s", err)
+		return "", fmt.Errorf("Error parsing API URL: %s", err)
 	}
 	u.Path = path.Join(u.Path, "1", "team_slug")
 	req, err := http.NewRequest("GET", u.String(), nil)
 	if err != nil {
-		return err
+		return "", err
 	}
 	req.Header.Set("User-Agent", UserAgentAddition)
 	req.Header.Add("X-Honeycomb-Team", config.WriteKey)
 	client := &http.Client{}
 	resp, err := client.Do(req)
 	if err != nil {
-		return err
+		return "", err
 	}
 	defer resp.Body.Close()
 	if resp.StatusCode == http.StatusUnauthorized {
-		return errors.New("Write key provided is invalid")
+		return "", errors.New("Write key provided is invalid")
 	}
+	body, _ := ioutil.ReadAll(resp.Body)
 	if resp.StatusCode != http.StatusOK {
-		body, _ := ioutil.ReadAll(resp.Body)
-		return fmt.Errorf(`Abnormal non-200 response verifying Honeycomb write key: %d
+		return "", fmt.Errorf(`Abnormal non-200 response verifying Honeycomb write key: %d
 Response body: %s`, resp.StatusCode, string(body))
 	}
+	ret := map[string]string{}
+	if err := json.Unmarshal(body, &ret); err != nil {
+		return "", err
+	}
 
-	return nil
+	return ret["team_slug"], nil
```

Comment:
hmm...  this part makes me a little nervous. what's the desired (or expected) behavior if the Honeycomb server sends back something that is JSON but doesn't have the `team_slug` key? We could fail more explicitly by trying to unmarshal the return body into a struct, or test for the key's existence then return. I think either way would get us a better error message for that case.

Reply:
Because `ret` is a `map[string]string`, if the server sends back JSON without that key, `ret["team_slug"]` will return an empty string. The purpose of this method is to verify the write key -- failing more explicitly based on the existence of a `"team_slug"` key doesn't feel in scope for this method (or this library, tbh).

Reply:
👍